backend/sales/models.py

```diff
@@ -0,0 +1,20 @@
+from django.db import models
+from books.models import Book
+from django.core.validators import MinValueValidator
+
+
+class SalesReconciliation(models.Model):
+    date = models.DateField()
+
+
+class Sale(models.Model):
+    book = models.ForeignKey(Book, related_name='books', on_delete=models.CASCADE)
+    quantity = models.PositiveIntegerField()
+    unit_retail_price = models.FloatField(validators=[MinValueValidator(0)])
+    sales_reconciliation = models.ForeignKey(SalesReconciliation, related_name='sales', on_delete=models.CASCADE)
+    revenue = models.FloatField()
+
+    def save(self, *args, **kwargs):
+        self.revenue = f'{self.quantity*self.unit_retail_price:.2f}'
```

Comment:
revenue seems to be a float field. Should revenue and retail_price be floats not strings?